lib/timers.js

```diff
@@ -496,7 +501,15 @@ function rearm(timer) {
     timer._handle.start(timer._repeat);
   } else {
     timer._idleTimeout = timer._repeat;
-    active(timer);
+
+    const duration = TimerWrap.now() - start;
+    if (duration >= timer._repeat) {
+      // If callback duration >= timer._repeat,
+      // add 1 ms to avoid blocking eventloop
+      process.nextTick(insert, timer, false, start + duration - timer._repeat);
+    } else {
+      insert(timer, false, start);
+    }
```

Comment:
I had meant the entire block here, I don't think the conditional or the +- start is necessary.

In an ideal world, replacing `active(timer)` with `process.nextTick(insert, timer, false, start)` would be all that were necessary. Unfortunately that causes a slight continuity break and the existing timers iteration code is unaware that a timer will be scheduled in the same slot but only after this iteration, causing the underlying TimerWrap to be re-created (and thus impacting AsyncHooks tests).

Imo, `insert(timer, false, start + 1)` is the way to go.